velox/exec/GroupingSet.cpp

```diff
@@ -458,6 +458,24 @@ void GroupingSet::resetPartial() {
   }
 }
 
+bool GroupingSet::isPartialFull(int64_t maxBytes) {
+  if (!table_ || allocatedBytes() <= maxBytes) {
+    return false;
+  }
+  if (table_->hashMode() == BaseHashTable::HashMode::kArray) {
+    auto stats = table_->stats();
+    // If we have a large array with sparse data, we rehash this in a
+    // mode that turns off value ranges for kArray mode. Large means
+    // over 1/16 of the space budget and sparse means under 1 entry
+    // per 32 buckets.
+    if (stats.capacity * sizeof(void*) > maxBytes / 16 &&
+        stats.numDistinct < stats.capacity / 32)
+      table_->decideHashMode(0, true);
+    return allocatedBytes() > maxBytes;
+  }
+  return true;
```

Comment:
Is this right? Shouldn't we return allocatedBytes() > maxBytes; ?

Reply:
@oerling 

> The return value of is..Full is right. Check at the start.

Got it. The way the code is written is confusing though. Consider changing to something like:

```
if (!table_ || allocatedBytes() <= maxBytes) {
    return false;
  }

if (table_->hashMode() != BaseHashTable::HashMode::kArray) {
    return true;
}

...
```